src/libraries/System.Text.Json/src/System/Text/Json/Serialization/Converters/Collection/IEnumerableConverterFactoryHelpers.cs

```diff
@@ -254,13 +254,32 @@ private static Type GetImmutableDictionaryConstructingType(Type type)
 
         public static bool IsNonGenericStackOrQueue(this Type type)
         {
-            Type? typeOfStack = Type.GetType("System.Collections.Stack, mscorlib, Version=4.0.0.0, Culture=neutral, PublicKeyToken=b77a5c561934e089");
-            Type? typeOfQueue = Type.GetType("System.Collections.Queue, mscorlib, Version=4.0.0.0, Culture=neutral, PublicKeyToken=b77a5c561934e089");
+#if BUILDING_INBOX_LIBRARY
+            // Optimize for linking scenarios where mscorlib is trimmed out.
+            string stackName = "System.Collections.Stack, System.Collections.NonGeneric, Version=4.0.0.0, Culture=neutral, PublicKeyToken=b77a5c561934e089";
```

Comment:
You do not really need the super long name for inbox library. It can be just "System.Collections.Stack, System.Collections.NonGeneric".

It is what we are doing in other places, e.g.
https://github.com/dotnet/runtime/blob/ed78fadab63e6f8c4ee227521be93ec3aece1dc9/src/coreclr/src/System.Private.CoreLib/src/System/Reflection/Emit/ILGenerator.cs#L1140